src/modules/commander/HealthAndArmingChecks/checks/estimatorCheck.cpp

```diff
@@ -273,25 +273,29 @@ void EstimatorChecks::checkEstimatorStatus(const Context &context, Report &repor
 		}
 
 		if (!context.isArmed() && ekf_gps_check_fail) {
-			NavModes required_groups_gps;
+			NavModesMessageFail required_modes;
 			events::Log log_level;
 
 			switch (static_cast<GnssArmingCheck>(_param_com_arm_wo_gps.get())) {
 			default:
 
 			/* FALLTHROUGH */
 			case GnssArmingCheck::DenyArming:
-				required_groups_gps = required_groups;
+				required_modes.message_modes = required_modes.fail_modes = NavModes::All;
 				log_level = events::Log::Error;
 				break;
 
 			case GnssArmingCheck::WarningOnly:
-				required_groups_gps = NavModes::None; // optional
+				required_modes.message_modes = (NavModes)(reporter.failsafeFlags().mode_req_local_position
+							       | reporter.failsafeFlags().mode_req_local_position_relaxed
+							       | reporter.failsafeFlags().mode_req_global_position);
+				// Don't block arming for the case there's a valid position estimate from another source e.g. vehicle with optical flow + GPS
```

Comment:
```suggestion
				// Don't block arming, only warn
```
I guess that's what you wanted to relay here? It doesn't even have to have valid local position, does it? With the existing comment it sounds to me like some specific implementation for "valid position estimate but no GPS".

Reply:
Let me change the wording.
https://github.com/PX4/PX4-Autopilot/compare/7300b57799f261e286fb3dbe5635904e3bff6981..940d7bde1a4478f8a37b6eaf2bb5a4a06bb38cc4#diff-61d9b9f86ef33be4a6b79cfa1df04ceb6c2464dc1179de837d661a23e71732afR292

> valid position estimate but no GPS

It's the case 5 (second last) in my table above:
Imagine a drone with GPS and optical flow flying indoor in position mode. It should show a warning for GPS reception problems but you should still be able to arm and fly in Position mode using optical flow.
Without this case we could also block arming since if GPS was the only source for a position we couldn't arm anyways.